### PR TITLE
Backport: Added the ability to freely change chat fonts + fixed dropdown overflow

### PR DESCRIPTION
**Pull request by Watermelon914**
## About The Pull Request

https://gitlab.com/cmdevs/colonial-warfare/-/merge_requests/593

Able to pick your own chat font from a list of fonts in chat settings
Pic1 image
Pic2 image
Pic3 image

Also fixed dropdown overflow so that it doesn't break if the selected input is too long. Instead, it cuts it off with an ellipsis.
image

## Why It's Good For The Game

More user friendly

## Changelog
:cl:
add: Added chat fonts that can be chosen via settings. You can also input a custom font.
fix: Fixed dropdown overflow if the selected input is too long.
/:cl:

### CODE REVIEW

**Review by stylemistake — APPROVED** [review]
Very good. I don't understand why dropdowns have this weird space at the bottom. Is it because of your dropdown changes in css?

Previous PR backported by spooky: #55928

**Comment by Watermelon914** [comment]
> Very good. I don't understand why dropdowns have this weird space at the bottom. Is it because of your dropdown changes in css?
> 
> Previous PR backported by spooky: #55928

No, it happened before and only happens when you open the dropdown for the first time.

**Comment by Watermelon914** [comment]
image
Fixed

**Comment by stylemistake** [comment]
Hmmm, baselines are misaligned, try adding `inline align="baseline"` to Stack.

**Comment by Watermelon914** [comment]
image
Nice

**Review by stylemistake — APPROVED** [review]
Not perfect, but I accept. 👍

**Comment by zxaber** [comment]
Is this VV-changeable? Can I punish players by setting their font to wingdings?

**Comment by Watermelon914** [comment]
> Is this VV-changeable? Can I punish players by setting their font to wingdings?

Nah, it's all done clientside.

**Comment by stylemistake** [comment]
You can forge and inject a tgui message as an admin that changes player's font.

**Comment by cacogen** [comment]
now everything can be comic sans

File: tgui/packages/tgui-panel/settings/constants.js [hunk 1]
```diff
@@ -14,3 +14,20 @@ export const SETTINGS_TABS = [
     name: 'Chat Tabs',
   },
 ];
+
+
+export const FONTS_DISABLED = "Use CSS";
```
Comment (Jared-Fogle) [inline tgui/packages/tgui-panel/settings/constants.js:19]:
```suggestion
export const FONTS_DISABLED = "Default";
```

I don't like this exposing of implementation details to players.

File: tgui/packages/tgui-panel/settings/SettingsPanel.js [hunk 2]
```diff
@@ -5,13 +5,14 @@
...
 import { toFixed } from 'common/math';
+import { useLocalState } from 'tgui/backend';
 import { useDispatch, useSelector } from 'common/redux';
 import { Box, Button, ColorBox, Divider, Dropdown, Flex, Input, LabeledList, NumberInput, Section, Stack, Tabs, TextArea } from 'tgui/components';
 import { ChatPageSettings } from '../chat';
 import { rebuildChat, saveChatToDisk } from '../chat/actions';
 import { THEMES } from '../themes';
 import { changeSettingsTab, updateSettings } from './actions';
-import { SETTINGS_TABS } from './constants';
+import { SETTINGS_TABS, FONTS } from './constants';
```
Comment (Jared-Fogle) [inline tgui/packages/tgui-panel/settings/SettingsPanel.js:15]:
```suggestion
import { FONTS, SETTINGS_TABS } from './constants';
```

File: tgui/packages/tgui-panel/settings/reducer.js [hunk 3]
```diff
@@ -5,11 +5,12 @@
  */
 
 import { changeSettingsTab, loadSettings, openChatSettings, toggleSettings, updateSettings } from './actions';
-import { SETTINGS_TABS } from './constants';
+import { SETTINGS_TABS, FONTS } from './constants';
```
Comment (Jared-Fogle) [inline tgui/packages/tgui-panel/settings/reducer.js:8]:
```suggestion
import { FONTS, SETTINGS_TABS } from './constants';
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Minor feedback.

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-5). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**